payments/models.py

```diff
@@ -676,36 +693,39 @@ def save(self, *args, **kwargs):
             unit = self.product.price_unit
 
-            if not self.price:
-                if unit == PriceUnits.PERCENTAGE:
-                    price = calculate_product_percentage_price(
-                        self.order.price, self.product.price_value
+            if unit == PriceUnits.PERCENTAGE:
+                price = calculate_product_percentage_price(
+                    self.order.price, self.product.price_value
+                )
+
+            if self.order.lease:
+                if self.product.period == PeriodType.MONTH:
+                    price = calculate_product_partial_month_price(
+                        price, self.order.lease.start_date, self.order.lease.end_date,
+                    )
+                elif self.product.period == PeriodType.SEASON:
+                    # Calculate the actual price for the amount of days on the period
+                    # price = (days_on_lease * product.price_value) / season_days
+                    price = calculate_product_partial_season_price(
+                        price,
+                        self.order.lease.start_date,
+                        self.order.lease.end_date,
+                        summer_season=isinstance(self.order.lease, BerthLease),
+                    )
+                elif self.product.period == PeriodType.YEAR:
+                    price = calculate_product_partial_year_price(
+                        price, self.order.lease.start_date, self.order.lease.end_date,
                     )
 
-                if self.order.lease:
-                    if self.product.period == PeriodType.MONTH:
-                        price = calculate_product_partial_month_price(
-                            price,
-                            self.order.lease.start_date,
-                            self.order.lease.end_date,
-                        )
-                    elif self.product.period == PeriodType.SEASON:
-                        # Calculate the actual price for the amount of days on the period
-                        # price = (days_on_lease * product.price_value) / season_days
-                        price = calculate_product_partial_season_price(
-                            price,
-                            self.order.lease.start_date,
-                            self.order.lease.end_date,
-                            summer_season=isinstance(self.order.lease, BerthLease),
-                        )
-                    elif self.product.period == PeriodType.YEAR:
-                        price = calculate_product_partial_year_price(
-                            price,
-                            self.order.lease.start_date,
-                            self.order.lease.end_date,
-                        )
+            if hasattr(self.order.customer, "organization"):
+                organization_type = self.order.customer.organization.organization_type
+
+                self.price = calculate_organization_price(price, organization_type)
```

Comment:
Not calling the `round_price` here

Reply:
It was not called here before, should it be called?

Reply:
I think so, it should be `rounded` right before being set to ensure that it keeps the `two rounded decimals` rule. If we update the called function, this can stay as-is.

Reply:
done